23_1.cpp

```diff
@@ -0,0 +1,59 @@
+#include <functional>
+#include <iostream>
+#include <queue>
+#include <vector>
+
+struct ListNode {
+  int val;
+  ListNode* next;
+  explicit ListNode(int x) : val(x), next(nullptr) {}
+};
+
+struct IdWithValue {
+  int id_;
+  int value_;
+  IdWithValue(int id, int value) : id_(id), value_(value) {}
+  bool operator>(IdWithValue other) const {
+    return value_ > other.value_;
+  }
+};
+
+class Solution {
+ public:
+  ListNode* mergeKLists(std::vector<ListNode*>& lists) {  // NOLINT
+    std::priority_queue<
+        IdWithValue, std::vector<IdWithValue>, std::greater<IdWithValue> >
+        heap;

```

Comment:
Note, you can also override `operator<` as
```C++
inline bool operator<(const IdWIthValue& lhs, const IdWithValue& rhs) const {
    return lhs.value > rhs.value;
}
```
and use 
```C++
    std::priority_queue<IdWithValue> heap;
```
(Your method also looks fine)